simapp/sim_test.go

```diff
@@ -168,84 +283,182 @@ func appStateRandomizedFn(r *rand.Rand, accs []simulation.Account, genesisTimest
-			simulation.ModuleParamSimulator["TallyParams/Veto"](r).(sdk.Dec),
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "tally_params_quorum", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["tally_params_quorum"](r).(sdk.Dec) })
+				return v
+			}(r),
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "tally_params_threshold", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["tally_params_threshold"](r).(sdk.Dec) })
+				return v
+			}(r),
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "tally_params_veto", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["tally_params_veto"](r).(sdk.Dec) })
+				return v
+			}(r),
 		),
 	)
+
+	fmt.Printf("Selected randomly generated governance parameters:\n%s\n", mustMarshalJSONIndent(cdc, govGenesis))
 	genesisState[gov.ModuleName] = cdc.MustMarshalJSON(govGenesis)
-	fmt.Printf("Selected randomly generated governance parameters:\n\t%+v\n", govGenesis)
+}
 
-	stakingGenesis := staking.NewGenesisState(
-		staking.InitialPool(),
-		staking.NewParams(
-			simulation.ModuleParamSimulator["UnbondingTime"](r).(time.Duration),
-			simulation.ModuleParamSimulator["MaxValidators"](r).(uint16),
-			7,
+func genMintGenesisState(cdc *codec.Codec, r *rand.Rand, ap simulation.AppParams, genesisState map[string]json.RawMessage) {
+	mintGenesis := mint.NewGenesisState(
+		mint.InitialMinter(
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "inflation", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["inflation"](r).(sdk.Dec) })
+				return v
+			}(r),
+		),
+		mint.NewParams(
 			sdk.DefaultBondDenom,
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "inflation_rate_change", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["inflation_rate_change"](r).(sdk.Dec) })
+				return v
+			}(r),
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "inflation_max", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["inflation_max"](r).(sdk.Dec) })
+				return v
+			}(r),
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "inflation_min", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["inflation_min"](r).(sdk.Dec) })
+				return v
+			}(r),
+			func(r *rand.Rand) sdk.Dec {
+				var v sdk.Dec
+				ap.GetOrGenerate(cdc, "goal_bonded", &v, r, func(r *rand.Rand) { v = simulation.ModuleParamSimulator["goal_bonded"](r).(sdk.Dec) })
+				return v
+			}(r),
+			uint64(60*60*8766/5),
+		),
+	)
+
+	fmt.Printf("Selected randomly generated minting parameters:\n%s\n", mustMarshalJSONIndent(cdc, mintGenesis.Params))
+	genesisState[mint.ModuleName] = cdc.MustMarshalJSON(mintGenesis)
+}
+
+func genDistrGenesisState(cdc *codec.Codec, r *rand.Rand, ap simulation.AppParams, genesisState map[string]json.RawMessage) {
+	// TODO: make use NewGenesisState
```

Comment:
would it be possible to address this TODO as part of this PR 🙏 ?

Reply:
Sure, updated. Although I think it reads much more and is part of my biggest gripe with being forced to use constructors.

Reply:
Actually, I'm not going to commit this but instead rather remove the TODO. I don't think this beneficial at all....a bulk of the args are `nil` anyway.